python_ta/transforms/setendings.py

```diff
@@ -191,6 +295,60 @@ def discover_nodes(node):
     logging.info(message)
 
 
+def fix_slice(source_code):
+    """
+    The Slice node column positions are mostly set properly when it has (Const) children, all it needs is to consume leading/trailing whitespace.
+
+    The main problem is when Slice node doesn't have children. E.g "[:]", "[::]", "[:][:]", or even "[::][::]", yikes! The existing col_offset of the slice node is set improperly to 0. And the end_col_offset is also wrong. We fix it here.
+    """
+    def _consume_subscripts(node):
+
+        if not node.last_child():
+            if not hasattr(node, 'end_lineno'): set_without_children(node)
+
+            line_i = node.parent.fromlineno - 1  # 1-based
+            char_i = node.parent.col_offset  # 0-based
+
+            # To solve the problem created by variations of "a[:][:]",
+            # all sibling Subscript node characters to consume.
+            sibling_buffer = ''
```

Comment:
Correction, this approach mentioned above actually *does* work, because all the transforms are applied to nodes as they are encountered as the AST is walked -- rather than one node type at-a-time.